services/venice-controller/src/main/java/com/linkedin/venice/pushmonitor/AbstractPushMonitor.java

```diff
@@ -149,30 +149,16 @@ private void loadAllPushes(List<OfflinePushStatus> offlinePushStatusList) {
     pushStatusCollector.start();
     try (AutoCloseableLock ignore = clusterLockManager.createClusterWriteLock()) {
       LOGGER.info("Load all pushes started for cluster {}'s {}", clusterName, getClass().getSimpleName());
-      // Subscribe to changes first
-      List<OfflinePushStatus> refreshedOfflinePushStatusList = new ArrayList<>();
       for (OfflinePushStatus offlinePushStatus: offlinePushStatusList) {
         try {
+          topicToPushMap.put(offlinePushStatus.getKafkaTopic(), offlinePushStatus);
```

Comment:
Yes, so I think about this - if you check listener callback, it will first read the OfflinePushStatus object from the local map, if you don't put it first (even if it might going to be a bit stale), it will result in change got dropped. So this is necessary.

Reply:
thanks for the explain, we can keep them

Reply:
Synced offline, we decided to keep it for now as safeguard, while we add store write lock for another EV listener callback method